package/gluon-core/luasrc/usr/lib/lua/gluon/wireless.lua

```diff
@@ -2,16 +2,45 @@ local sysconfig = require 'gluon.sysconfig'
 local site = require 'gluon.site'
 local util = require 'gluon.util'
 
+local unistd = require 'posix.unistd'
+local dirent = require 'posix.dirent'
+
 
 local M = {}
 
 local function find_phy_by_path(path)
-	local phy = util.glob('/sys/devices/' .. path .. '/ieee80211/phy*')[1]
-		or util.glob('/sys/devices/platform/' .. path .. '/ieee80211/phy*')[1]
+	local device_path, phy_offset
+	-- Special handling required for multi-phy devices
+	if string.match(path, '+%d$') then
+		device_path, phy_offset = string.match(path, "^(.+)%+(%d+)$")
+
+		-- The UCI radio path is split into the path and a PHY offset,
+		-- which is incremented for each additional PHY available on
+		-- the device.
+	else
+		device_path = path
+		phy_offset = 0
```

Comment:
There is something wrong here - in the `if` branch, `phy_offset` is a string, while it is a number in the `else` branch.

Also, calling `string.match` twice is unnecessary - just try the split and check if `device_path` is nil afterwards (and the first check is still limited to single-digit indices).